tests/request-response/request_response_client_tests.c

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+#include <aws/mqtt/private/request-response/protocol_adapter.h>
+#include <aws/mqtt/request-response/request_response_client.h>
+
+#include <aws/testing/aws_test_harness.h>
+
+#include "../v3/mqtt311_testing_utils.h"
+#include "../v5/mqtt5_testing_utils.h"
+
+enum rr_test_client_protocol {
+    RRCP_MQTT311,
+    RRCP_MQTT5,
+};
+
+struct aws_rr_client_test_fixture {
+    struct aws_allocator *allocator;
+
+    struct aws_mqtt_request_response_client *client;
```

Comment:
Trivial: For a little added clarity in tests, can we rename this to `rr_client` or something that explicitly separates this from the mqtt_client that is also being referenced in tests?